### DISCUSSION TITLE
Setting to prevent SumatraPDF from editing annotations, or Undo function

### DISCUSSION THREAD

**Discussion by danfingerman** [sumatrapdfreader/sumatrapdf#2462] [Ideas]
I have sometimes accidentally edited (moved) a preexisting annotation that I did not intend to edit. Could a setting be created to disable editing annotations? Ideally, the user should be able to set the state whether annotations can be edited or not without leaving the interface. 

As an alternative, an Undo function would be useful. Currently, the only way to undo an edit is to close the PDF and discard all changes. That is not ideal, because other (intended) changes should be saved.

**Comment by GitHubRulesOK** [comment]
It as been raised/discussed in https://github.com/sumatrapdfreader/sumatrapdf/discussions/2331 and a related [open issue](https://github.com/sumatrapdfreader/sumatrapdf/issues/1681) is linked there

**Comment by kjk** [comment]
Undo is a possibility. I don't foresee adding option to disable annotation editing.

**Comment by sumsu** [comment]
It's very annoying if you try to copy text from file, where is a lot of annotations. With disabled annotations everything is text which you can copy. With enabled annotations you have to move aside annotations to dig to lowest layer for text what are you looking for.
